Documentation on various config parameters.

aaruna opened the pull request:
Also added info on CQL types supported for Source

aaruna commented on README.md:
```diff
@@ -51,15 +41,81 @@ SELECT * FROM userlog WHERE ts >= previousTime() AND  ts <= currentTime() ; //ti
+| cassandra.connection.keep_alive (not yet implemented) | Period of time, in ms, to keep unused connections open. |  |
+| cassandra.connection.reconnect.delay.min | Minimum period of time (in ms) to wait before reconnecting to a dead node. | 1000 |
+| cassandra.connection.reconnect.delay.max | Maximum period of time (in ms) to wait before reconnecting to a dead node. | 60000 |
+| cassandra.connection.compression (broken) | One of the compressor strategies. Can be DeflateCompressor, SnappyCompressor or LZ4Compressor | NONE |
```
I found this broken. 
Since we are doing ` ProtocolOptions.Compression.valueOf(s)` in https://github.com/tuplejump/kafka-connect-cassandra/blob/DB-53/src/main/scala/com/tuplejump/kafka/connect/cassandra/CassandraCluster.scala#L213 ,  the possible values should be `snappy` or `lz4` or ``(empty)

helena replied:
I don't have a test for that yet, thank you for the catch.

helena replied:
I'll create a ticket and fix later this week.

aaruna commented on README.md:
```diff
@@ -51,15 +41,81 @@ SELECT * FROM userlog WHERE ts >= previousTime() AND  ts <= currentTime() ; //ti
+| | **Source Connect Properties** | |
+| cassandra.source.route.\<topic_name\> | The Select Query to get the data. (Refer CassandraSource documentation for more details) | |_
+| cassandra.source.poll.interval | Frequency in ms to poll for new data in each table. | 60000 |
+| cassandra.source.split.size (not yet implemented?) | Size of Cassandra data to be read in a single Kafka task;  determines the number of partitions. | 64 |
```
@helena What is this used for?

helena replied:
@aaruna nothing yet, will be useful with paging. If not I'll remove.

aaruna commented on README.md:
```diff
@@ -51,15 +41,81 @@ SELECT * FROM userlog WHERE ts >= previousTime() AND  ts <= currentTime() ; //ti
+| cassandra.source.split.size (not yet implemented?) | Size of Cassandra data to be read in a single Kafka task;  determines the number of partitions. | 64 |
+| cassandra.source.fetch.size | Number of CQL rows to fetch in a single round-trip to Cassandra. | 1000 |
+| cassandra.source.consistency (not yet implemented?) | Consistency level for reads, defaults to LOCAL_ONE;  higher consistency level will disable data-locality.| LOCAL_ONE |
+| cassandra.source.timezone (not yet implemented?) | ?? | true |
```
@helena What is this used for?

helena replied:
If the user wants system, i.e. local time or UTC. It is used. @Shiti originally has java system time, this allows the user to decide.

aaruna replied:
This config also is not being used, so removing it from Readme for now. We can add it back once we fix it

aaruna commented on README.md:
```diff
@@ -51,15 +41,81 @@ SELECT * FROM userlog WHERE ts >= previousTime() AND  ts <= currentTime() ; //ti
+| cassandra.source.poll.interval | Frequency in ms to poll for new data in each table. | 60000 |
+| cassandra.source.split.size (not yet implemented?) | Size of Cassandra data to be read in a single Kafka task;  determines the number of partitions. | 64 |
+| cassandra.source.fetch.size | Number of CQL rows to fetch in a single round-trip to Cassandra. | 1000 |
+| cassandra.source.consistency (not yet implemented?) | Consistency level for reads, defaults to LOCAL_ONE;  higher consistency level will disable data-locality.| LOCAL_ONE |
```
We are never using this value. Instead we are always picking the read consistency level from `cassandra.connection.consistency`

helena replied:
Yes, this is also not implemented yet but I want the user to be able to pass in specific config per table/query if they need it. `cassandra.connection.consistency` is just a default if not set explicitly (coming soon).

Shiti commented on README.md:
```diff
@@ -51,15 +41,81 @@ SELECT * FROM userlog WHERE ts >= previousTime() AND  ts <= currentTime() ; //ti
+cassandra.connection.host=10.0.0.1
+cassandra.connection.port=9042
+
+cassandra.source.route.topic1=keyspace1.table1
```
this should be `cassandra.sink.route.topic1`

Shiti commented:
@aaruna @helena This documentation is very lengthy. Can we add index with links or move it to github wiki?

helena commented:
Week will add a proper doc plugin to the build and partition it out. This ticket is to address accuracy given the new configurations and config additions. Item to discuss which plugin. Wiki is a no - no revision ties to version/branch.

aaruna commented:
I have added ToC and removed all unused config parameters, so that the release is not blocked. We can add the additional parameters as and when we fix them.
